[trivial] Fixing different typos and grammar issues

dramarereg opened the pull request:
## Description

This pull request addresses typos and minor corrections in documentation files. Specifically, the changes fix spelling errors and clarify wording in the following files:

1. **README.md** - Corrected "lienar" to "linear."
2. **aggregator_v2.md** - Fixed "uncoditional" to "unconditional."
3. **code.md** - Corrected "publised" to "published."
4. **config_buffer.md** - Fixed "sychronous" to "synchronous" and "on-chaon" to "on-chain."
5. **dispatchable_fungible_asset.md** - Corrected "guranteed" to "guaranteed."

These changes improve the clarity and readability of the documentation.

---

## How Has This Been Tested?

- Verified the changes manually to ensure corrected typos do not impact formatting or context.

---

## Key Areas to Review

- Confirm that the corrections align with the intended context of the documentation.
- Ensure that no formatting or reference links were inadvertently affected.

---

## Type of Change
- [x] Documentation update

---

## Which Components or Systems Does This Change Impact?
- [x] Documentation

---

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [x] I have made corresponding changes to the documentation

---

georgemitenkov reviewed (changes requested):
Hey @dramarereg! Thanks for contribution, few notes:

1. For PR name, please name it in a way it is easy to understand what is this, not just `Fix`. A good example in this case would be `[trivial] Fixing different typos and grammar issues`, or something like that.
2. Markdown files for framework code documentation (e.g., `aggregatror_v2.md`) are automatically generated. To really fix the typos you need to modify corresponding `.move` files, e.g., https://github.com/aptos-labs/aptos-core/blob/7f1f7df848bce1c91251855e22c123c994e9921d/aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.move#L26

I am not sure how doc generation is triggered, but usually running linter will do:
```
cargo +nightly fmt && cargo xclippy && cargo sort --grouped --check --workspace
```

dramarereg commented:
Hi, @georgemitenkov! Thank you for the feedback.

I have completed all the points except the last one:

1. Updated the PR title to `[trivial] Fixing different typos and grammar issues`.
2. Fixed typos directly in the `.move` files, including `aptos-core/aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.move` and other files.
3. Updated comments in `aggregator_v2.move` and similar files where necessary. For example:
   ```rust
   /// The value of the aggregator overflows. Raised by an unconditional add() call.

Unfortunately, I couldn’t run the linter with the following command:

```bash
cargo +nightly fmt && cargo xclippy && cargo sort --grouped --check --workspace

georgemitenkov reviewed (approved):
Looks good now, thanks! Linter doesn't matter (it just also triggers automatic doc generation iirc), the important thing is to make sure .move files are also changed (otherwise your changed to .md get reverted)
